### PR TITLE
Basic block styling & blur functionality

### PR DESCRIPTION
**Pull request by Markvanderuit**
If anyone has any issues with the blur at all, say so immediatly.

### CODE REVIEW

File: src/main/java/gui/TimetableBlock.java [hunk 1]
```diff
@@ -81,67 +152,166 @@
...
+        Label res = new Label();
+        res.setText("0 - 0");
+        res.maxWidthProperty().bind(this.widthProperty());
+        res.setPadding(new Insets(0,5,5,5));
+        res.setStyle(textNormalStyle);
+        vbox.getChildren().add(res);
+        return res;
     }
 
+//    /**
```
Comment (Markvanderuit) [inline src/main/java/gui/TimetableBlock.java]: Ehh whoops, I'll remove that in the morning

File: src/main/java/control/TimelineController.java [hunk 2]
```diff
@@ -17,6 +16,8 @@
 import javafx.stage.FileChooser.ExtensionFilter;
 import lombok.Getter;
 
+import java.io.File;
```
Comment (bart-de-jonge) [inline src/main/java/control/TimelineController.java]: java.io.File is imported twice

File: src/main/java/gui/BlurHelper.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+ * Also has support for moving elements!
+ */
+public class BlurHelper {
+
+    private WritableImage writableImage;
+    private SnapshotParameters parameters;
+    private Bounds bounds;
+
+    @Getter
+    private GaussianBlur gaussianBlur;
```
Comment (bart-de-jonge) [inline src/main/java/gui/BlurHelper.java]: You could add some comments to these fields to explain what they do

File: src/main/java/gui/CameraShotBlock.java [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import data.CameraShot;
 import javafx.event.EventHandler;
+import javafx.scene.control.Label;
```
Comment (bart-de-jonge) [inline src/main/java/gui/CameraShotBlock.java]: Is this import ever used in this class?

File: src/main/java/gui/ShotBlock.java [hunk 5]
```diff
@@ -28,7 +28,7 @@
     private String description;
 
     // The name of this shotblock
-    @Getter @Setter
+    @Getter
```
Comment (bart-de-jonge) [inline src/main/java/gui/ShotBlock.java:31]: Why did you remove the setter here?

File: src/main/java/gui/TimelinesGridPane.java [hunk 6]
```diff
@@ -44,13 +44,13 @@ public TimelinesGridPane(int numberOfTimelines, int numberOfCounts,
...
-        this.setMaxHeight(height);
+        this.setMinHeight(countHeight * numberOfCounts);
+        this.setMaxHeight(countHeight * numberOfCounts);
 
         addPanes();
 
-        this.setGridLinesVisible(true);
-        this.setPadding(new Insets(5, 5, 5, 5));
+        this.setGridLinesVisible(false);
+        //this.setPadding(new Insets(5, 5, 5, 5)); // don't remove this. - Mark
```
Comment (bart-de-jonge) [inline src/main/java/gui/TimelinesGridPane.java]: Maybe you could explain why this should not be removed

File: src/main/java/gui/BlurHelper.java (hunk 3 above)
Comment (Markvanderuit) [inline src/main/java/gui/BlurHelper.java]: I could and I shall,, but its boilerplate code, so I don't see the point.

File: src/main/java/gui/CameraShotBlock.java (hunk 4 above)
Comment (Markvanderuit) [inline src/main/java/gui/CameraShotBlock.java]: Nope. removed.

File: src/main/java/gui/ShotBlock.java (hunk 5 above)
Comment (Markvanderuit) [inline src/main/java/gui/ShotBlock.java:31]: Because I define setName() lower in the class, as it also does extra stuff now. (update name of the timetableblock)

File: src/main/java/gui/TimetableBlock.java [hunk 7]
```diff
@@ -5,68 +5,139 @@
...
+    private String colorBlockBackground = "#B1E2FA";
+    private String colorBlockBorders = "#DDF2FC";
+    private String colorBlockBorderHighlight = "#005482";
+    private String colorText = "#047FB7";
+
+    /*
+        Temporary variables, until they can be moved to CSS files properly.
+     */
+
+    private String normalStyleBackground =
```
Comment (bart-de-jonge) [inline src/main/java/gui/TimetableBlock.java:55]: Is it possible to extract these style rules to different classes or maybe even files to prevent cluttering this class? This might also be done in future PR's

File: src/main/java/gui/TimetableBlock.java (hunk 1 above)
Comment (bart-de-jonge) [inline src/main/java/gui/TimetableBlock.java]: Did you already remove this?

File: src/main/java/gui/BlurHelper.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    @Getter @Setter
+    private Node node;
+
+    @Getter @Setter
+    private boolean hideNode;
+
+    @Getter @Setter
+    private Point2D offset;
+
+
```
Comment (martijnjanssen) [inline src/main/java/gui/BlurHelper.java:42]: Not needed.

**Comment by bart-de-jonge** [comment]
Other then the comments it looks good, please look at those. On a sidenote: committing often is important, but 16 commits for one PR is a little bit too much in my opinion. Could you seperate functionality like the basic styling and the blurring into seperate PR's to really keep a PR about one type of functionality at a time.

File: src/main/java/gui/BlurHelper.java [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        gaussianBlur = new GaussianBlur(radius);
+        parameters = new SnapshotParameters();
+        imageView = new ImageView();
+        imageView.setEffect(gaussianBlur);
+        offset = new Point2D(0,0);
+    }
+
+    /**
+     * Radius setter, that also updates the blur effect radius.
+     * @param radius new radius to use/
```
Comment (martijnjanssen) [inline src/main/java/gui/BlurHelper.java]: Redundant `/`.

File: src/main/java/gui/BlurHelper.java [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+    /**
+     * Process blur for this BlurHelper, by using the bounds of the node.
+     * WARNING: if the node is being dragged or resized, make sure this is called AFTER
+     * the node has moved or resized. It may prevent jittering!
+     */
+    public void processBlurUsingBounds() {
+        // get information on current size of object in scene, and set proper parameters
+        bounds = node.localToScene(node.getBoundsInLocal());
+        Rectangle2D rekt = new Rectangle2D(bounds.getMinX() + offset.getX() ,
```
Comment (martijnjanssen) [inline src/main/java/gui/BlurHelper.java]: Please change this variable name to the correct spelling. The correct spelling is `rect`.

File: src/main/java/gui/BlurHelper.java (hunk 3 above)
Comment (bart-de-jonge) [inline src/main/java/gui/BlurHelper.java]: One line is enough, just for documentation purposes. More comments = more better

File: src/main/java/gui/ShotBlock.java (hunk 5 above)
Comment (bart-de-jonge) [inline src/main/java/gui/ShotBlock.java:31]: My bad

File: src/main/java/gui/RootPane.java [hunk 11]
```diff
@@ -40,7 +40,9 @@ public void start(Stage primaryStage) throws Exception {
         topLevelPane = new BorderPane();
         // Create scene and set the stage. This is where the window is basically
         // created. Also has some useful settings.
-        primaryStage.setScene(new Scene(topLevelPane));
+        Scene scene = new Scene(topLevelPane);
+        scene.getStylesheets().add("testcss.css");
```
Comment (martijnjanssen) [inline src/main/java/gui/RootPane.java:45]: Please add a `TODO:` to change this line of code.

File: src/main/java/gui/ShotBlock.java [hunk 12]
```diff
@@ -101,6 +102,34 @@ public void setEndCount(double count) {
     }
 
     /**
+     * Set name of this shotblock. No Lombok because it does some extra work for the GUI.
```
Comment (martijnjanssen) [inline src/main/java/gui/ShotBlock.java:105]: The reason this setter is not generated by Lombok does not have to be mentioned explicitly.

File: src/main/java/gui/TimetableBlock.java [hunk 13]
```diff
@@ -5,68 +5,139 @@
...
 import javafx.scene.layout.Region;
 import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
 import javafx.scene.shape.Rectangle;
 import lombok.Getter;
 
 /**
- * Class that resembles a draggable, resiable block inside the timetable.
- * Highly volatile. Do not poke too much.
+ * Class that resembles a draggable, resiable block inside the timetable,
```
Comment (martijnjanssen) [inline src/main/java/gui/TimetableBlock.java]: `resizable`

File: src/main/java/gui/TimetableBlock.java [hunk 14]
```diff
@@ -5,68 +5,139 @@
...
  */
 public class TimetableBlock extends Pane {
 
     public enum DraggingTypes { Move, Resize_Top, Resize_Right, Resize_Bottom, Resize_Left }
 
     /*
         Styling variables.
-        For styling and displayable content.
+        For tweaking the styling.
      */
```
Comment (martijnjanssen) [inline src/main/java/gui/TimetableBlock.java:40]:
``` java
/**
 * Styling variables.
 * For tweaking the styling.
 */
```

File: src/main/java/gui/TimetableBlock.java [hunk 15]
```diff
@@ -81,67 +152,166 @@
...
+        clipRegion.heightProperty().bind(pane.heightProperty().subtract(verticalBorderSize));
+        content.setClip(clipRegion);
+        pane.getChildren().add(content);
     }
 
     /**
-     * Temporary helper function to add test labels to panes.
+     * Helper function to add title labels to panes.
      * @param vbox pane to add to
+     * @return the label in question.
```
Comment (martijnjanssen) [inline src/main/java/gui/TimetableBlock.java:261]: The `@param` line does not look complete.

File: src/main/java/gui/TimetableBlock.java [hunk 16]
```diff
@@ -81,67 +152,166 @@
...
+        res.setPadding(new Insets(0,0,0,5));
+        res.setStyle(textTitleStyle);
+        vbox.getChildren().add(res);
+        return res;
+    }
+
+    /**
+     * Helper function to add count labels to panes.
+     * @param vbox pane to add to
+     * @return the label in question.
```
Comment (martijnjanssen) [inline src/main/java/gui/TimetableBlock.java:276]: Same here, the `@param` tag.

File: src/main/java/gui/TimetableBlock.java [hunk 17]
```diff
@@ -81,67 +152,166 @@
...
+//     */
+//    private void addTestLabels(VBox vbox) {
+//        for (int i = 0; i < 6; i++) {
+//            Label label = new Label(shots);
+//            label.maxWidthProperty().bind(this.widthProperty());
+//            label.setPadding(new Insets(5,5,5,5));
+//            label.setStyle(textNormalStyle);
+//            vbox.getChildren().add(label);
+//        }
+//    }
```
Comment (martijnjanssen) [inline src/main/java/gui/TimetableBlock.java]: Why is all this code commented?

File: src/main/java/gui/TimetableBlock.java (hunk 1 above)
Comment (Markvanderuit) [inline src/main/java/gui/TimetableBlock.java]: Removed.

File: src/main/java/gui/TimetableBlock.java (hunk 17 above)
Comment (Markvanderuit) [inline src/main/java/gui/TimetableBlock.java]: Already gone.

**Comment by bart-de-jonge** [comment]
Great work, merging
